tests/app/user/test_rest.py

```diff
@@ -252,22 +231,20 @@ def test_cannot_create_user_with_empty_strings(admin_request, notify_db_session)
     ('email_address', 'newuser@mail.com'),
     ('mobile_number', '+4407700900460')
 ])
-def test_post_user_attribute(client, sample_user, user_attribute, user_value):
+def test_post_user_attribute(admin_request, sample_user, user_attribute, user_value):
     assert getattr(sample_user, user_attribute) != user_value
     update_dict = {
         user_attribute: user_value
     }
-    auth_header = create_admin_authorization_header()
-    headers = [('Content-Type', 'application/json'), auth_header]
 
-    resp = client.post(
-        url_for('user.update_user_attribute', user_id=sample_user.id),
-        data=json.dumps(update_dict),
-        headers=headers)
+    json_resp = admin_request.post(
+        'user.update_user_attribute',
+        user_id=sample_user.id,
+        _data=update_dict
+    )
 
-    assert resp.status_code == 200
-    json_resp = json.loads(resp.get_data(as_text=True))
     assert json_resp['data'][user_attribute] == user_value
+    assert getattr(sample_user, user_attribute) == user_value
```

Comment:
I was confused to see this addition at first but I think I get it: we want to check the model was actually updated, not just that the POSTed data is echoed back in the response.